std/digest/murmurhash.d

```diff
@@ -0,0 +1,860 @@
+/**
+Computes MurmurHash hashes of arbitrary data. MurmurHash is a non-cryptographic
+hash function suitable for general hash-based lookup. It is optimized for x86
+but can be used on every architectures.
+
+The current version is MurmurHash3, which yields a 32-bit or 128-bit hash value.
+The older MurmurHash 1 and 2 are currently not supported.
+
+MurmurHash3 comes in three flavors, providing greater and greater throughput:
```

Comment:
providing greater and greater throughput -> listed in increasing order of throughput